src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs

```diff
@@ -280,6 +367,39 @@ private static NameNode GenerateInputTypeName(string typeName)
                 new NamedTypeNode(returnEntityName),
                 fieldDefinitionNodeDirectives
             );
+
+            // Multiple insertion node.
+            FieldDefinitionNode createMultipleNode = new(
+                location: null,
+                new NameNode($"create{GetInsertMultipleMutationName(singularName, GetDefinedPluralName(name.Value, entity))}"),
+                new StringValueNode($"Creates multiple new {singularName}"),
+                new List<InputValueDefinitionNode> {
+                new(
+                    location : null,
+                    new NameNode(ARRAY_INPUT_ARGUMENT_NAME),
+                    new StringValueNode($"Input representing all the fields for creating {name}"),
+                    new ListTypeNode(new NonNullTypeNode(new NamedTypeNode(input.Name))),
+                    defaultValue: null,
+                    new List<DirectiveNode>())
+                },
+                new NamedTypeNode(QueryBuilder.GeneratePaginationTypeName(GetDefinedSingularName(dbEntityName, entity))),
+                fieldDefinitionNodeDirectives
+            );
+
+            return new(createOneNode, createMultipleNode);
+        }
+
+        /// <summary>
+        /// Helper method to determine the name of the insert multiple mutation.
+        /// If the singular and plural graphql names for the entity match, we suffix the name with the insert multiple mutation suffix.
```

Comment:
"we suffix the name with the insert multiple mutation suffix." can you add to the sentence with "because *reason*". e.g. avoid naming collisions.

Reply:
Yeah the reason is to indicate that the mutation field is created to support insertion of multiple records in the top level entity. Updating summary.